### PR TITLE
Bump snapshot-controller and snapshotter deploy to v5-rc1

### PR DESCRIPTION
**Pull request by ggriffiths**
Signed-off-by: Grant Griffiths <ggriffiths@purestorage.com>

**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
Needed to make tests pass

**Which issue(s) this PR fixes**:
n/a

**Special notes for your reviewer**:
n/a

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

**Comment by ggriffiths** [comment]
cc @xing-yang

File: deploy/kubernetes/csi-snapshotter/setup-csi-snapshotter.yaml [hunk 1]
```diff
@@ -84,7 +84,7 @@ spec:
             - name: socket-dir
               mountPath: /csi
         - name: csi-snapshotter
-          image: k8s.gcr.io/sig-storage/csi-snapshotter:v4.2.1
+          image: gcr.io/k8s-staging-sig-storage/csi-snapshotter:v5.0.0-rc1
```
Comment (xing-yang) [inline deploy/kubernetes/csi-snapshotter/setup-csi-snapshotter.yaml:87]: Can you add a TODO here to replace this with the official image once v5.0.0 is released?

File: deploy/kubernetes/snapshot-controller/setup-snapshot-controller.yaml [hunk 2]
```diff
@@ -33,7 +33,7 @@ spec:
       serviceAccount: snapshot-controller
       containers:
         - name: snapshot-controller
-          image: k8s.gcr.io/sig-storage/snapshot-controller:v4.2.1
+          image: gcr.io/k8s-staging-sig-storage/snapshot-controller:v5.0.0-rc1
```
Comment (xing-yang) [inline deploy/kubernetes/snapshot-controller/setup-snapshot-controller.yaml:36]: Same as above.

File: deploy/kubernetes/csi-snapshotter/setup-csi-snapshotter.yaml (hunk 1 above)
Comment (ggriffiths) [reply deploy/kubernetes/csi-snapshotter/setup-csi-snapshotter.yaml:89]: Done

File: deploy/kubernetes/snapshot-controller/setup-snapshot-controller.yaml (hunk 2 above)
Comment (ggriffiths) [reply deploy/kubernetes/snapshot-controller/setup-snapshot-controller.yaml:38]: Done

**Pull request merged**